### PR TITLE
Add web project tests

### PR DESCRIPTION
**Pull request by a-roberts**
# Changes

This is for https://github.com/tektoncd/dashboard/issues/90. Print Node.js and npm versions then run basic web tests including linting.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

Let's see what this gives us, definitely the right file to change, testing this now and will check if we need changes to plumbing too.

Update: boo

```
I0503 10:19:59.116] ./test/presubmit-tests.sh: line 198: npm: command not found
I0503 10:19:59.116] >> Node.js version
I0503 10:19:59.117] ./test/presubmit-tests.sh: line 200: node: command not found
```

### CODE REVIEW

**Comment by a-roberts** [comment]
@AlanGreene @steveodonovan I've made some progress, see https://prow.tekton.dev/log?job=pull-tekton-dashboard-tests&id=1124265258249097216

See my super WIP code at https://github.com/a-roberts/dashboard/blob/webtests/test/presubmit-tests.sh.

@vdemeester are we cool to `sudo` our commands? 

I see the script is running as root. 

I've curled dependencies I want (Node.js which includes npm) directly and modified the PATH variable, IDK if there's an official way to install deps through Prow (where do we get Go from?) so I'm holding off on this until I know.

**Comment by a-roberts** [comment]
@AlanGreene I spoke with Vincent [here](https://tektoncd.slack.com/archives/CJ4ERJWAU/p1556888021011300), I'll create an issue for having our own test-infra image (we're currently using the knative one so we just get go). Have a look at the versions I'm pulling in etc please and lemme know if you're happy to get this in

File: test/presubmit-tests.sh [hunk 1]
```diff
@@ -108,6 +108,28 @@ function initialize_environment() {
...
+  tar xf node-v10.15.3-linux-x64.tar.xz
+  export PATH=$PATH:$(pwd)/node-v10.15.3-linux-x64/bin
+}
+
+function node_test() {
+  echo "Running node tests from $(pwd)"
+  mkdir ~/.npm-global
+  npm config set prefix '~/.npm-global'
+  export PATH=$PATH:$HOME/.npm-global/bin
+  npm install -g --unsafe-perm node-sass
```
Comment (AlanGreene) [inline test/presubmit-tests.sh]: We shouldn't need to install any packages globally, and node-sass is already specified in package.json so seems a bit odd to have this. Were you hitting a particular issue without it?

Comment (a-roberts) [reply test/presubmit-tests.sh]: In an old commit yeah, then I added the npm config fun stuff (it was permission denied to do with node_modules folder and things running as `nobody` presumably)

Just removed it and all is well, have a look at https://prow.tekton.dev/log?job=pull-tekton-dashboard-tests&id=1125687504725348352 😄 (the latest Prow run for this PR)

**Comment by a-roberts** [comment]
@AlanGreene PTAL again and approve/lgtm if you're happy, I've removed the unnecessary dependency install and everything's fine, thanks

**Pull request merged**